crates/matrix-sdk/src/account.rs

```diff
@@ -231,12 +231,11 @@ impl Account {
     /// # block_on(async {
     /// # let homeserver = Url::parse("http://localhost:8080")?;
     /// # let client = Client::new(homeserver).await?;
-    /// if let profile = client.account().get_profile().await? {
```

Comment:
Hmm, the intention here might have been to have an `if let Some(profile) = ...` line. Since a profile might be `None`.

Reply:
It's caught by `?`.

So either we write:

```rust
if let Some(profile) = ….get_profile().await
```

or we write:

```
let profile = ….get_profile().await?
```

but we have to choose :-p. In any case, `if let profile = …` is already true, so the `if` is useless.

Reply:
Ah no, I was wrong, this returns a `Result<FooResponse>`, not a `Result<Option<FooProfile>>`. You're right, the `?` handles the `Result`.